pkg/services/auth/auth.go

```diff
@@ -58,10 +58,19 @@ type RevokeAuthTokenCmd struct {
 	AuthTokenId int64 `json:"authTokenId"`
 }
 
+type RotateCommand struct {
+	// token is the un-hashed token
+	UnHashedToken string
+	IP            net.IP
```

Comment:
Why do we need IP and UserAgent ?

Reply:
They are stored in the table along side the tokens. I added a new function to perform rotation instead of reusing the old one, TryRotateToken, to always force a rotation when called